### PR TITLE
[BUGFIX] fix valid candidates issue

### PR DESCRIPTION
**Pull request by liuzh91**
## Description ##
Fix the dynamic mask sampling strategy. Fix Issue #1321 

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [x] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

cc @dmlc/gluon-nlp-team

### CODE REVIEW

File: scripts/pretraining/pretraining_utils.py [hunk 1]
```diff
@@ -499,16 +499,16 @@ def dynamic_masking(self, F, input_ids, valid_lengths):
...
         valid_lengths = valid_lengths.astype(np.float32)
         valid_candidates = valid_candidates.astype(np.float32)
         num_masked_position = F.np.maximum(
             1, F.np.minimum(N, round(valid_lengths * self._mask_prob)))
 
         # Get the masking probability of each position
-        sample_probs = F.npx.softmax(
-            self._proposal_distribution * valid_candidates, axis=-1)  # (B, L)
+        sample_probs = self._proposal_distribution * valid_candidates
+        sample_probs /= np.sum(sample_probs, axis=-1, keepdims=True)
```
Comment (sxjscience) [inline scripts/pretraining/pretraining_utils.py:511]: Use sample_probs = sample_probs / F.np.sum(sample_probs, axis=-1, keepdims=True)

**Comment by sxjscience** [comment]
@liuzh91 Would you try to fix this along with the other Gumbel-topk issue? (So there is no need for me to merge twice).

**Comment by liuzh91** [comment]
> @liuzh91 Would you try to fix this along with the other Gumbel-topk issue? (So there is no need for me to merge twice).


OK, I'll work on that
